syntax = "proto3";

import "priority.proto";
import "date.proto";
import "task_id.proto";

message TaskProto {
  string              name = 1;
  string              label = 2;
  ProtoPriority      priority = 3;
  ProtoDate          deadline = 4;
  bool                completed = 5;
  repeated TaskProto  subtasks = 6;
  ProtoTaskID        parent_id = 7;
}

message StorageProto {
  repeated TaskProto  tasks = 1;
}